src/response/mod.rs

```diff
@@ -16,35 +16,43 @@ pub struct Response {
 
 impl Response {
     pub fn new(status: HttpStatus) -> Self {
+        let mut headers = Headers::new();
+        // Initialize Content-Length: 0 so headers are always stored in the struct,
+        // not created on-the-fly during write().
+        headers.set_content_length(0);
+
         Self {
             status,
-            headers: Headers::new(),
+            headers,
             body: None,
         }
     }
 
     pub fn set_str_body(&mut self, body: &str) {
         let bytes = body.as_bytes();
         self.headers.set("Content-Type", "text/plain");
-        self.headers.set("Content-Length", &bytes.len().to_string());
-        self.body = Some(Vec::from(body.as_bytes()));
+        self.headers.set_content_length(bytes.len());
+        self.body = Some(Vec::from(bytes));
```

Comment:
Use `HttpBody` instead of `Option<Vec<u8>>`. The `HttpBody` is already an enum and can represent both empty and content.